modules/determining-upgrade-viability-conditiontype.adoc

```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
+//
+// * updating/index.adoc
+
+:_content-type: CONCEPT
+[id="determining-upgrade-viability-conditiontype_{context}"]
+= Determining upgrade viability with conditionType
+
+The status of cluster operators is revealed with the value of conditionType, informing you of potential impacts to upgrading your cluster. 
+
+* Available: 
+The conditionType `Available` indicates that the operators are functional and available in the cluster. If the status is `Available=False`, then the status indicates that at least one part of the operand is non-functional and the condition requires an administrator to intervene.
+
+* Progressing:
+The conditionType `Progressing` indicates that the operator is actively rolling out new code, propagating configuration changes, or otherwise moving from one steady state to another. Operators should not report as `Progressing` when they are reconciling a previous known state. If the observed cluster state has changed and the operator is reacting to it, then the status should report back as `Progressing=True`, since it is moving from one steady state to another.
```

Comment:
> Operators should not report as `Progressing` when they are reconciling a previous known state.

The "should" here and in the following sentences are inherited [from godocs][1], and seem like they are encouraging operator maintainers to pay attention.  But the context in these new docs, where we're trying to explain how readers consume the fields and not also talking to operator-maintainers setting them, we might want a more imperative:

> Operators report `Progressing=False` when they are reconciling...

and similar.  And if a condition consumer notices an operator violating the nominal imperative (e.g. with `Progressing!=False` while reconciling a previous known state), they can open a bug against that operator.

[1]: https://github.com/openshift/api/blob/cce310ad2932f6de24491052d506926e484c082c/config/v1/types_cluster_operator.go#L152-L158